subscriber.go

```diff
@@ -270,13 +273,7 @@ func NewRedisSubscriber(address string, handler SubscriptionHandler, poolSize in
 		slots:       make([]*redisSubscriberConn, poolSize),
 	}
 	for slot := 0; slot < poolSize; slot++ {
-		// XXX comment from strib: do we have to block on these
-		// subscription calls?  I think the caller might find it
-		// unexpected.  Plus, something that blocks should probably
-		// take a context that can be cancelled by the caller if
-		// things take too long.  Either way, maybe we should document
-		// the blocking behavior in the method comment?
-		subscriber.reconnectSlot(slot, 0)
+		go subscriber.reconnectSlot(slot, 0)
```

Comment:
Oh, you might be right.  I couldn't reproduce it without closures, though the text of that section in effective go is ambiguous about it.